Assets/Scripts/NPCs/StateMachine/NPCStateMachine.cs

```diff
@@ -40,10 +37,11 @@ private void Update()
     
     /*
      * If the player collides with a NPC that is not chasing him, start chase.
+     * For now, the collider is of type trigger because there is no navmesh.
      */
-    public void CollidedWithPlayer()
+    public void OnTriggerEnter(Collider collider)
     {
-        if (_currentState != _chase)
+        if (_currentState != _chase && collider.gameObject.CompareTag("Player"))
         {
             GetComponent<AudioSource>().PlayOneShot(bumpingSounds[0]);
             GetComponent<Animator>().SetTrigger(Animator.StringToHash("Get Hit"));
```

Comment:
Thoses `GetComponent` calls will be repeated each time, move them to a class variable

Reply:
Those are only called once per NPC. I guess I can put them in variable anyway.

Reply:
Oh that's true, I forgot the if condition